graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java

```diff
@@ -117,38 +135,60 @@ public boolean bulkIndex(final List<Message> messages) {
         return !response.hasFailures();
     }
 
-    private void propagateFailure(BulkItemResponse[] items, List<Message> messages, String errorMessage) {
-        log.error(
-                "Failed to index [{}] messages. Please check the index error log in your web interface for the reason. Error: {}",
-                items.length,
-                errorMessage);
+    private BulkResponse runBulkRequest(final BulkRequest request) {
+        try {
+            return c.bulk(request).actionGet();
+        } catch (ElasticsearchTimeoutException timeoutException) {
+            LOG.debug("Bulk indexing request timed out. Retrying.", timeoutException);
+            try {
+                return BULK_REQUEST_RETRYER.call(new BulkRequestCallable(c, request));
```

Comment:
AFAIK the `BulkRequest` itself (the structure being passed to the Elasticsearch client which then performs the actual remote call) doesn't cache or resolve anything on construction. So this should be fine™.